src/Tribe/Views/V2/Views/Traits/HTML_Cache.php

```diff
@@ -0,0 +1,192 @@
+
+namespace Tribe\Events\Views\V2\Views\Traits;
+
+use Tribe__Cache_Listener as Cache_Listener;
+use Tribe__Context as Context;
+use Tribe__Date_Utils as Dates;
+
+/**
+ * Trait HTML_Cache
+ *
+ * @since   5.0.0
+ *
+ * @package Tribe\Events\Views\V2\Views
+ *
+ * @property Context $context The current View context.
+ */
+trait HTML_Cache {
+	/**
+	 * Return cached HTML if enabled and cache is set.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return null|string
+	 */
+	public function maybe_get_cached_html() {
+
+		if ( ! $this->should_cache_html() ) {
+			return;
+		}
+
+		$cache_key = $this->get_cache_html_key();
+
+		if ( $cached_html = tribe( 'cache' )->get_transient( $cache_key, $this->cache_html_triggers() ) ) {
+			return $cached_html;
+		}
+	}
+
+	/**
+	 * If caching is enabled, set the cache.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param string $html HTML markup for view.
+	 */
+	public function maybe_cache_html( $html ) {
+		/**
+		 * Filter the cache TTL
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param int     $cache_ttl Cache time to live.
+		 * @param Context $context   The View current context.
+		 * @param View    $this      The current View instance.
+		 */
+		$cache_expiration = apply_filters( 'tribe_events_views_v2_cache_html_expiration', DAY_IN_SECONDS, $this->get_context(), $this );
+
+		$cache_key = $this->get_cache_html_key();
+
+		if ( ! $this->should_cache_html() ) {
+			return;
+		}
+
+		tribe( 'cache' )->set_transient( $cache_key, $html, $cache_expiration, $this->cache_html_triggers() );
+	}
+
+	/**
+	 * Fetch the HTML cache triggers.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return array
+	 */
+	protected function cache_html_triggers() {
+		$triggers = [
+			Cache_Listener::TRIGGER_SAVE_POST,
+			Cache_Listener::TRIGGER_UPDATED_OPTION,
+			Cache_Listener::TRIGGER_GENERATE_REWRITE_RULES,
+		];
+
+		return $triggers;
+	}
+
+	/**
+	 * Determine if HTML of the current view needs to be cached.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return bool
+	 */
+	public function should_cache_html() {
+		$context = $this->get_context();
+
+		$cached_views = [
+			'month',
+			'week',
+		];
+
+		$should_cache = in_array( $this->get_slug(), $cached_views ) && $this->should_enable_html_cache( $context );
+
+		/**
+		 * Should the v2 view HTML be cached?
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param bool    $should_cache_html Should the current view have its HTML cached?
+		 * @param Context $context           The View current context.
+		 * @param View    $this              The current View instance.
+		 */
+		return (bool) apply_filters( 'tribe_events_views_v2_should_cache_html', $should_cache, $context, $this );
+	}
+
+	/**
+	 * Determine if HTML of the current view needs to be cached.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @return bool
+	 */
+	public function get_cache_html_key() {
+		$context = $this->get_context();
+		$args    = $context->to_array();
+
+		unset( $args['now'] );
+
+		$cache_key = 'tribe_views_v2_cache_' . substr( sha1( json_encode( $args ) ), 0, 12 ) . ':';
+
```

Comment:
:warning: json_encode() is discouraged. Use wp_json_encode() instead.

<sub>hash: 97982d876e428aca55f893be4d484ccb</sub>